frappe/core/doctype/user/user.js

```diff
@@ -114,12 +114,16 @@ frappe.ui.form.on("User", {
 			return;
 		}
 
+		function hasChanged(doc_attr, boot_attr) {
+			return (doc_attr || boot_attr) && doc_attr !== boot_attr;
```

Comment:
@gavindsouza if `doc_attr` is undefined, we get stuck in an endless refresh loop.

This would fix it:

```js
return doc_attr !== undefined && (doc_attr || boot_attr) && doc_attr !== boot_attr;
```

Reply:
when does this happen?

seems to work fine for me - even when creating a new user with language field not set. apparently we have `""` as default value rather than undefined.

![Screenshot from 2023-03-28 11-06-58](https://user-images.githubusercontent.com/32034600/228140097-9492849b-71b9-4b9b-a84f-9ea9dfa3536b.png)

This is what it was before as well - just been taken out into a function

Reply:
Undefined seemed to work for we. But maybe check for empty string or truthiness as well. When the fields are empty and didn't change, we don't want to reload.